rholang/src/main/scala/coop/rchain/rholang/interpreter/ContractCall.scala

```diff
@@ -29,24 +30,26 @@ import cats.implicits._
   */
 class ContractCall[F[_]: Sync](
     space: RhoISpace[F],
-    dispatcher: Dispatch[F, ListParWithRandomAndPhlos, TaggedContinuation]
+    dispatcher: Dispatch[F, ListParWithRandom, TaggedContinuation]
 ) {
   type Producer = (Seq[Par], Par) => F[Unit]
 
-  implicit val cost: _cost[F] =
-    loggingCost(CostAccounting.unsafe[F](Cost(Integer.MAX_VALUE)), noOpCostLog)
-
+  // TODO: pass _cost[F] as an implicit parameter
   private def produce(
       rand: Blake2b512Random,
       sequenceNumber: Int
   )(values: Seq[Par], ch: Par): F[Unit] =
     for {
-      produceResult <- space.produce(
-                        ch,
-                        ListParWithRandom(values, rand),
-                        persist = false,
-                        sequenceNumber
-                      )(matchListPar)
+      costAlg <- CostAccounting.of(Cost(0))
+      cost    = loggingCost(costAlg, noOpCostLog[F])
+      produceResult <- noCharge {
```

Comment:
having 
```scala
noCharge { costAlg =>
  implicit val cost = loggingCost(costAlg, noOpCostLog[F])
  ...
}
```
we could make it clear that the only part doing any cost accounting is inside `noCharge`, and have one line of code less :sunglasses: (not counting braces :))

Reply:
I like that. I'd rather make those changes in a subsequent PR though. I'd like to get a `noCharge` operation that didn't require instantiating new instances of `CostAccounting` and `_cost` 🤔